src/ORM/FieldType/DBDate.php

```
@@ -58,35 +61,34 @@ public function setValue(mixed $value, null|array|ModelData $record = null, bool
      * Parse timestamp or iso8601-ish date into standard iso8601 format
      *
      * @param mixed $value
-     * @return string|null|false Formatted date, null if empty but valid, or false if invalid
+     * @return mixed Formatted date, or the original value if it couldn't be parsed
      */
-    protected function parseDate(mixed $value): string|null|false
+    protected function parseDate(mixed $value): mixed
     {
-        // Skip empty values
-        if (empty($value) && !is_numeric($value)) {
-            return null;
-        }
```

Why do we not skip empty values anymore?

---

If you pass in nonsense like an empty array or an object, it should fail validation, rather than gracefully being  converted to null

---

Reverted